aws-lc-fips-sys/builder/cmake_builder.rs

```diff
@@ -137,11 +156,31 @@ impl CmakeBuilder {
             .configure_arg("--no-warn-unused-cli")
             .build()
     }
+
+    fn collect_vcvarsall_bat(&self) -> Result<HashMap<String, String>, String> {
+        let mut map: HashMap<String, String> = HashMap::new();
+        let script_path = self.manifest_dir.join("builder").join("printenv.bat");
+        let result = test_command(script_path.as_os_str(), &[]);
+        if !result.status {
+            return Err("Failed to run vccarsall.bat.".to_owned());
+        }
+        let lines = result.output.lines();
+        for line in lines {
+            if let Some((var, val)) = line.split_once('=') {
+                map.insert(var.to_string(), val.to_string());
+            }
+        }
+        Ok(map)
+    }
 }
 
 impl crate::Builder for CmakeBuilder {
     fn check_dependencies(&self) -> Result<(), String> {
         let mut missing_dependency = false;
+        if target_os() == "windows" && !test_ninja_command() {
+            eprintln!("Missing dependency: Ninja is required for FIPS on Windows.");
```

Comment:
is this true for non-FIPS as well? looks like we don't check FIPSness on L180

Reply:
The `aws-lc-fips-sys` crate only gets built for FIPS.